### ISSUE TITLE
Dark theme support: Make coloring of svg icons independent of Mozilla specific context switches 

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplate4#310]
Without the switch `svg.context-properties.content.enabled = true` enabled svg icons on the SmartTemplates preferences screens are hard to read in dark mode:

![image](https://github.com/user-attachments/assets/0b43cf0b-025d-4f45-a328-614d1d5b27a1)

It also affects some of the follow up action icons:

![image](https://github.com/user-attachments/assets/4f64d6f9-fa47-4453-ac1d-6b6257c92f09)

In order to make theming colors more robust, let's instead include the color variables within the svg and use a media query to trigger the view from within.

**Comment by RealRaven2000** [comment]
So here is an example for a svg image used by Thunderbird (leaving out Mozilla license comment for brevity)

```svg
<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
  <path fill="context-fill" fill-opacity="context-fill-opacity" d="M11.5 10a2.481 2.481 0 0 0-.379.038L3.977 1.214a2.5 2.5 0 0 0-.371 3.515l2.789 3.444-1.51 1.866A2.486 2.486 0 0 0 4.5 10a2.522 2.522 0 1 0 2.329 1.609L8 10.159 9.172 11.6A2.5 2.5 0 1 0 11.5 10zm-7 3.75a1.25 1.25 0 1 1 1.25-1.25 1.251 1.251 0 0 1-1.25 1.25zm7 0a1.25 1.25 0 1 1 1.25-1.25 1.251 1.251 0 0 1-1.25 1.25zm.9-9.021a2.5 2.5 0 0 0-.371-3.515L8.5 5.569l1.608 1.986z"/>
</svg>
```

I extended the icons as follows by adding a style block and (in this case) removing `fill="context-fill"` and `fill-opacity="context-fill-opacity"` .

Here is the added style block:

```svg
  <style>
  :root {
    --basiccolor: rgb(20,20,20);
    --fill: var(--basiccolor);
    --fill-opacity: 0.9;
  }
  @media (prefers-color-scheme: dark) {
    :root {
      --basiccolor: rgb(251,251,254);
      --fill: var(--basiccolor);
      --fill-opacity: 0.8;
    }
  }
  </style>
  <path 
    style="fill:var(--basiccolor); fill-opacity:var(--fill-opacity);"
    d="M11.5 10a2.481 2.481 0 0 0-.379.038L3.977 1.214a2.5 2.5 0 0 0-.371 3.515l2.789 3.444-1.51 1.866A2.486 2.486 0 0 0 4.5 10a2.522 2.522 0 1 0 2.329 1.609L8 10.159 9.172 11.6A2.5 2.5 0 1 0 11.5 10zm-7 3.75a1.25 1.25 0 1 1 1.25-1.25 1.251 1.251 0 0 1-1.25 1.25zm7 0a1.25 1.25 0 1 1 1.25-1.25 1.251 1.251 0 0 1-1.25 1.25zm.9-9.021a2.5 2.5 0 0 0-.371-3.515L8.5 5.569l1.608 1.986z"/>
```

note that I am using no media rule as default (which should cover light themes) and overwrite them for dark mode. The path tag then includes fill attributes that can use these variables.

**Comment by RealRaven2000** [comment]
Here is a preview version with the new icons:

[smartTemplate-fx-4.7.1pre13.zip](https://github.com/user-attachments/files/16570693/smartTemplate-fx-4.7.1pre13.zip)

-----

To install, download archive and drop into Thunderbird Add-ns Manager (without extracting contents).

**Comment by MarkyMarkDE** [comment]
yep, work as expected

**Comment by MarkyMarkDE** [comment]
@RealRaven2000 have you created the svg files byself or are this out of the box?

**Comment by RealRaven2000** [comment]
> @RealRaven2000 have you created the svg files byself or are this out of the box?

I created some of them myself, but the write / reply / forward / support (chat) and Global Settings are variations from the Thunderbird Core app. But I removed the "context" attributes because they are not standard (and bound to fail in html without the special switch or if / when Mozilla drops support)

**Comment by MarkyMarkDE** [comment]
> > @RealRaven2000 have you created the svg files byself or are this out of the box?
> 
> I created some of them myself, but the write / reply / forward / support (chat) and Global Settings are variations from the Thunderbird Core app. But I removed the "context" attributes because they are not standard (and bound to fail in html without the special switch or if / when Mozilla drops support)

ah okay, nice work.
By reason that the Pro Icon is being cropped off, maybe you can change this icon to a 16x16px golden crown icon or something ...

![Bildschirmfoto vom 2024-08-11 19-33-21](https://github.com/user-attachments/assets/5fe2ea15-d235-498f-89cb-2f4fecf95da9)

**Comment by RealRaven2000** [comment]
very strange - looks like the wrong font also. Maybe I need to include that one somehow (It's called Bahnschrift) didn't know this wasn't embedded in the scv automatically. Good catch.

**Comment by RealRaven2000** [comment]
it should look like this:

![image](https://github.com/user-attachments/assets/0bc16827-b055-4fb0-8fd8-bca18ba0dd0d)

**Comment by MarkyMarkDE** [comment]
> it should look like this:
> 
> ![image](https://private-user-images.githubusercontent.com/6314674/356903832-0bc16827-b055-4fb0-8fd8-bca18ba0dd0d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0MTAwMTIsIm5iZiI6MTcyMzQwOTcxMiwicGF0aCI6Ii82MzE0Njc0LzM1NjkwMzgzMi0wYmMxNjgyNy1iMDU1LTRmYjAtOGZkOC1iY2ExOGJhMGRkMGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTFUMjA1NTEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzI1NGM5M2VmMjY5NDI1NzFlZDg5NDBmNWM3MDBiMjFhNGFlZTZiOTI1ZGQxODQyZGU0MTIwZjcyNDFlY2UyYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.YlcJ74xOZWFdocHoHUoQ8Wqdye_uOsL3coLyawX4veM)

It should but didn't 😉.
The "I" icon and the pro icon looks completely different here as on your screenshot. Maybe it would be better, to replace this font based icon and replace it with a "graphical" one.
By the way: have you changed the font for the SmartTemplates Pro banner? Looks also funny here.

**Comment by RealRaven2000** [comment]
> The "I" icon and the pro icon looks completely different here as on your screenshot. Maybe it would be better, to replace this font based icon and replace it with a "graphical" one.

yeah I thought inkscape would do this automatically but I researched it yesterday and apparently there is way of converting fonts to path, so gotta do that with all the variations of the License icons, and any (i)nformation ones, too.

> By the way: have you changed the font for the SmartTemplates Pro banner? Looks also funny here.

I don't think so... but it may be that the original png got lost? It looks like this on Windows10:

![image](https://github.com/user-attachments/assets/4ea57c42-1aeb-4399-b1dc-420072df12b5)

**Comment by RealRaven2000** [comment]
@MarkyMarkDE see also my comments on QuickFolders (I thought I appended them here but don't want to duplicate it)

https://github.com/RealRaven2000/QuickFolders/issues/489#issuecomment-2283500070

**Comment by RealRaven2000** [comment]
Here is a version with converted icons (I turned all text icons to paths so they should look the same everywhere, independent on which are installed or not):

[smartTemplate-fx-4.7.1pre16.zip](https://github.com/user-attachments/files/16591686/smartTemplate-fx-4.7.1pre16.zip)

---

To install, download archive and drop into Thunderbird Add-ns Manager (without extracting contents).

**Comment by MarkyMarkDE** [comment]
the Pro Icon is great now, but the "i" icon ... see:
![Bildschirmfoto vom 2024-08-12 22-48-46](https://github.com/user-attachments/assets/690f3713-00c9-4cc2-8d21-da23d2d12a8a)

**Comment by RealRaven2000** [comment]
how about now:

[smartTemplate-fx-4.7.1pre19.zip](https://github.com/user-attachments/files/16592176/smartTemplate-fx-4.7.1pre19.zip)

looks like this in windows:

![image](https://github.com/user-attachments/assets/c1499ed8-5487-44af-9165-13748d159621)

you can save space with the screenshot if you make the Thunderbird window smaller.

**Comment by MarkyMarkDE** [comment]
> how about now:
> 
> [smartTemplate-fx-4.7.1pre19.zip](https://github.com/user-attachments/files/16592176/smartTemplate-fx-4.7.1pre19.zip)
> 
> looks like this in windows:
> 
> ![image](https://private-user-images.githubusercontent.com/6314674/357209937-c1499ed8-5487-44af-9165-13748d159621.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0OTk4MjIsIm5iZiI6MTcyMzQ5OTUyMiwicGF0aCI6Ii82MzE0Njc0LzM1NzIwOTkzNy1jMTQ5OWVkOC01NDg3LTQ0YWYtOTE2NS0xMzc0OGQxNTk2MjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTJUMjE1MjAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmFmMmM5ZGQyOWRlOTQ1NWFlM2Q3OTQwODVmY2VkM2I2NzljOTVmMzYwMDc3OTdkN2E4ODdlMmIwYTJmYjg5NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.FO8bVcZHsfGUu11achbSawAt2kZQ5HpB587ElNxyB-8)
> 
> you can save space with the screenshot if you make the Thunderbird window smaller.

now all is fine!
![Bildschirmfoto vom 2024-08-12 23-51-57](https://github.com/user-attachments/assets/39729005-f6dc-4fd7-a8bb-a620ebbb5a69)

**Comment by RealRaven2000** [comment]
I might add a nicer color scheme for the top icon (account templates) and Variables for dark mode while I am at it. I have figured out how to do this now, just need to target the correct elements (fill / stroke) to make it work...

**Comment by RealRaven2000** [comment]
More color tweaks in this one. For variables in dark mode I just need to figure out how to change the gradient in visual studio without messing it up, inkscape makes a very weird / complicated structure. Once I know how I will darken the green gradient.

[smartTemplate-fx-4.7.1pre24.zip](https://github.com/user-attachments/files/16592481/smartTemplate-fx-4.7.1pre24.zip)

![image](https://github.com/user-attachments/assets/da7e2e64-8cf0-4c65-933f-73d4756423eb)

**Comment by RealRaven2000** [comment]
OK, final version below. Fixed the graident on "Variables" in dark mode and applied similar stylinh to "Examples" as in Account templates. So this should be not such an eye sore in dark mode anymore:

![image](https://github.com/user-attachments/assets/2ced0118-d302-4500-8dad-0d749584727d)

[smartTemplate-fx-4.7.1pre31.zip](https://github.com/user-attachments/files/16598661/smartTemplate-fx-4.7.1pre31.zip)

---

To install, download archive and drop into Thunderbird Add-ns Manager (without extracting contents).

**Comment by RealRaven2000** [comment]
Implemented in 4.8 - published 03/09/2024
